service/history/execution/state_rebuilder.go

```diff
@@ -162,8 +162,10 @@ func (r *stateRebuilderImpl) Rebuild(
 		baseLastEventID,
 		baseLastEventVersion,
 	)) {
-		return nil, 0, &shared.InternalServiceError{Message: fmt.Sprintf(
-			"nDCStateRebuilder unable to rebuild mutable state to event ID: %v, version: %v",
+		return nil, 0, &shared.BadRequestError{Message: fmt.Sprintf(
```

Comment:
It makes sense to be a non retryable error. I understand the use of a bad request here but the bad request is more for user request. Could you add a TODO to mention have retryable error and non-retryable error?